domain/secretbackend/type.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package secretbackend
+
+import (
+	"time"
+
+	"github.com/juju/worker/v4"
+
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/core/secrets"
+	"github.com/juju/juju/core/watcher"
+)
+
+// CreateSecretBackendParams are used to create a secret backend.
+type CreateSecretBackendParams struct {
```

Comment:
Elsewhere, we've used params.go for this file; it contains params to pass to service calls.
We should be consistent.